text/trace/0157-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,184 @@
+the term "message" in a wider sense to cover both concepts.
+
+### Producer
+
+The "producer" is a specific instance, process or device that creates and
+publishes a message. "Publishing" is the process of sending a message or batch
+to the intermediary or consumer.
+
+### Consumer
+
+A "consumer" receives the message and acts upon it. It uses the context and
+data to execute some logic, which might lead to the occurrence of new events.
+
+The consumer receives, processes, and settles a message. "Receiving" is the
+process of obtaining a message from the intermediary, "processing" is the
+process of acting on the information a message contains, "settling" is the
+process of notifying an intermediary that a message was processed successfully.
+
+### Intermediary
+
+An "intermediary" receives a message to forward it to the next receiver, which
+might be another intermediary or a consumer.
+
+## Scenarios
+
+Producing and consuming a message involves five stages:
+
+```
+PRODUCER
+
+Create
+  |                            CONSUMER
+  v        +--------------+                   
+Publish -> | INTERMEDIARY | -> Receive
+           +--------------+       |
+                  ^               v
+                  .            Process
+                  .               |
+                  .               v
+                  . . . . . .  Settle
+```
+
+1. The producer creates a message.
+2. The producer publishes the message to an intermediary.
+3. The consumer receives the message from an intermediary.
+4. The consumer processes the message.
+5. The consumer settles the message by notifying the intermediary that the
+   message was processed successfully.
+
+The semantic conventions need to define how to handle failures and retries in
+all stages that interface with the intermediary: publish, receive and settle.
+
+Based on this model, the following scenarios capture major requirements and
+can be used for prototyping, as examples, and as test cases.
+
+### Individual settlement
+
+Individual settlement systems imply independent logical message flows. A single
+message is created and published in the same context, and it's delivered,
+consumed, and settled as a single entity. Each message needs to be settled
+individually.
+
+Transport batching can be treated as a special case: messages can be
+transported together as an optimization, but are produced and consumed
+individually.
+
+```
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+|Message A| |Message B| |Message C| |Message D| |Message E| |Message F|
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+ Settled                 Settled                             Settled 
+```
+
+#### Examples
+
+1. RabbitMQ: TODO
+
+### Checkpoint-based settlement
+
+Checkpoint-based settlement systems are likely to consume multiple messages at
+once. Messages are created, published (possibly) in batches with other
```

Comment:
I'm not sure about this sentence.

With Kafka, my understanding is publishing would usually not be batched and if there is a manual batching being done I'm not sure it can be done with unrelated messages. I think it would only be possible to batch messages that were being sent to the same logical destination, which would make the messages of the same type but different payloads

Reply:
You're right, it's misleading. I removed the reference to publishing in batches.